File: bootstrap/roles/testbench/files/code/sample.lua
-- comment; TODO: builtins, operators
local M = {}

local function greet(name, times)
  local out = ""
  for i = 1, times or 3 do
    out = out .. string.format("hi %s\n", name)
  end
  if name == nil or #out == 0 then
    print(out)
  end
  return { text = out, n = times }
end

M.greet = greet
return M

File: bootstrap/roles/testbench/files/code/sample.js
// TODO: builtins, template strings, operators
const MAX = 10;

export function greet(name = "world", times = 3) {
  const parts = [];
  for (let i = 0; i < times && i < MAX; i++) {
    parts.push(`hi ${name}\n`);
  }
  return parts.length === 0 ? null : parts.join("");
}

console.log(greet(process.argv[2]));
